files/build_templates/docker_image_ctl.j2

```diff
@@ -119,6 +116,17 @@ start() {
 
 {%- if docker_container_name == "database" %}
     echo "Creating new {{docker_container_name}} container"
+    # database_config.json contains the information how to create database instances and distribute the database on instances
+    # if file exist, use it, otherwise use the one in old system or create a dummy one 
```

Comment:
> if file exist, use it, otherwise use the one in old system or create a dummy one [](start = 5, length = 81)

We should assume it is exist, and readonly.

Reply:
if the database_config file is static as you said, then it always exists

Reply:
Yes, we assume it always exists. Could you add a default one in this PR?

---
In reply to: [240916318](https://github.com/Azure/sonic-buildimage/pull/2182#discussion_r240916318) [](ancestors = 240916318)

Reply:
Let me clarify.
1. This script will have no knowledge of `/etc/sonic/old_config` at all
2. Any image will provide a default `/etc/sonic/database_config.json`, same as a default `/etc/sonic/sonic_version.yml`
3. Service could fail if there is anything wrong with the file, or file missing.

Reply:
> Let me clarify.
> 
> 1. This script will have no knowledge of `/etc/sonic/old_config` at all
> 2. Any image will provide a default `/etc/sonic/database_config.json`, same as a default `/etc/sonic/sonic_version.yml`
> 3. Service could fail if there is anything wrong with the file, or file missing.

I know your point. But last time when we discuss this multi DB design durning Community meeting, Guohan prefer to have the ability to change the database_config.json file and then reload to make it take effect. That's why I introduce this part of codes. If we don't have the cold reboot case, then this is not necessary. You can double check with guohan and make sure our understanding is correct.

Reply:
This comment still apply.

---
In reply to: [271109776](https://github.com/Azure/sonic-buildimage/pull/2182#discussion_r271109776) [](ancestors = 271109776)